lib/View/welcome_screen.dart

```diff
@@ -0,0 +1,149 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'internallogin_screen.dart';
+import 'package:gocast_mobile/main.dart';
+
+class WelcomeScreen extends ConsumerWidget {
+  const WelcomeScreen({super.key});
+  Future<void> handleSSOLogin(
+      BuildContext context,
+      WidgetRef ref,
+      TextEditingController usernameController,
+      TextEditingController passwordController,
+      ) async {
+    // Call the SSO authentication function from /base/api/auth
+    await ref.read(userViewModel).ssoAuth(context);
+    // Navigate to the home screen after successful authentication
+    if (ref.read(userViewModel).current.value.user != null) {
+      // ignore: use_build_context_synchronously
+      await Navigator.pushNamed(context, '/welcome');
+    }
+  }
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
```

Comment:
The build methode is quite big consider breaking it int smaller readable widgets like:
- _buildHeader(...)
- _buildLoginButtons(...)
- _buildIndicatorDots()

Reply:
done